overlord/snapstate/snapstate.go

```diff
@@ -166,10 +174,18 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 
 	var prepare, prev *state.Task
 	fromStore := false
+
 	// if we have a local revision here we go back to that
 	if snapsup.SnapPath != "" || revisionIsLocal {
 		prepare = st.NewTask("prepare-snap", fmt.Sprintf(i18n.G("Prepare snap %q%s"), snapsup.SnapPath, revisionStr))
 	} else {
+		// XXX: DownloadInfo should never be nil, except for missing mocking in tests?
+		if snapsup.DownloadInfo != nil {
+			requiredSpace := requiredSpaceWithMargin(uint64(snapsup.DownloadInfo.Size))
+			if err := osutilCheckFreeSpace(dirs.SnapdStateDir(dirs.GlobalRootDir), requiredSpace); err != nil {
```

Comment:
I am a bit confused by the place for this.  If we have a refresh that touches many snaps or a more rare multi-install, shouldn't we consider the total size of all the downloads together one/some level ups?

Reply:
Yes, you are right. And it gets tricky with prerequisites / bases because we only know their names that early. We need to discuss that.